sapi/cli/php_cli_server.c

```diff
@@ -643,54 +674,67 @@ static void sapi_cli_server_register_variables(zval *track_vars_array) /* {{{ */
 			port[7] = '\0';
 			strncpy(addr, addr_start, addr_end - addr_start);
 			addr[addr_end - addr_start] = '\0';
-			sapi_cli_server_register_variable(track_vars_array, "REMOTE_ADDR", addr);
-			sapi_cli_server_register_variable(track_vars_array, "REMOTE_PORT", port);
+			sapi_cli_server_register_known_var_char(track_vars_array,
+				"REMOTE_ADDR", strlen("REMOTE_ADDR"), addr, strlen(addr));
```

Comment:
Can we pass `addr_end - addr_start` to avoid computing the string length at runtime?